Assets/src/Components/Wall/WallSystem.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Assets.src.Enums;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Assets.src.Components.Wall
+{
+    public class WallSystem : ComponentSystem
+    {
+        private struct Group
+        {
+            public Transform Transform;
+            public WallData WallData;
+            public WallPosition Position;
+            public MeshRenderer Renderer;
+        }
+
+        protected override void OnUpdate()
+        {
+            foreach (var entity in GetEntities<Group>())
+            {
+                entity.Transform.position = GetPositionVector(
+                    entity.WallData,
+                    entity.Position.RowPosition,
+                    entity.Position.ColumnPosition
+                );
+                entity.Transform.rotation = GetRotationQuaternion(entity.WallData);
+
+                entity.Renderer.material = GetMaterial(entity.WallData);
+            }
+        }
+
+        private Vector3 GetPositionVector(WallData wallData, float rowPos, float columnPos)
+        {
+            var yPos = wallData.IsActive ? 0.01F : -0.95F;
+            Vector3 result;
+
+            switch (wallData.Direction)
+            {
+                case DirectionEnum.Top:
+                    result = new Vector3(
```

Comment:
Maybe creating of vector should be function with parameters? In this case will be less repeatings in code?